packages/simplr-forms-core/__tests__/stores/form-store.test.ts

```diff
@@ -0,0 +1,153 @@
+import { FormStore } from "../../src/stores/form-store";
+import { FormError } from "../../src/contracts/error";
+
+describe("Form store", () => {
+    it("returns state", () => {
+        const formId = "FORM-ID";
+        const formStore = new FormStore(formId);
+
+        expect(formStore.GetState()).not.toBeUndefined();
+    });
+
+    it("returns fieldId from fieldName and fieldGroupId", () => {
+        const formId = "FORM-ID";
+        const fieldName = "FIELD-NAME";
+        const fieldGroupId = "FIELD-GROUP-ID";
+        const formStore = new FormStore(formId);
+
+        const fieldId = formStore.GetFieldId(fieldName, fieldGroupId);
+
+        expect(typeof fieldId).toBe("string");
+        expect(fieldId.indexOf(fieldName)).not.toBe(-1);
+        expect(fieldId.indexOf(fieldGroupId)).not.toBe(-1);
+    });
+
+    it("registers a field", () => {
+        const formId = "FORM-ID";
+        const fieldId = "FIELD-ID";
+        const initialValue = "INITIAL-VALUE";
+        const formStore = new FormStore(formId);
+
+        formStore.RegisterField(fieldId, initialValue);
+        expect(formStore.GetField(fieldId)).not.toBeUndefined();
+        expect(formStore.HasField(fieldId)).toBe(true);
+        expect(formStore.GetField(fieldId).InitialValue).toBe(initialValue);

```

Comment:
InitialValue is processed before saving it to store, which might change it and fail this test.